inc/namespace.php

```diff
@@ -326,6 +331,35 @@ function ajax_query_attachments() : void {
 	wp_send_json_success( $items->toArray() );
 }
 
+function ajax_get_attachment() {
+	if ( ! isset( $_REQUEST['id'] ) ) {
+		wp_send_json_error();
+	}
+
+	try {
+		$provider = ProviderRegistry::instance()->get( wp_unslash( $_REQUEST['provider'] ?? '' ) );
+		if (
+			strpos( $_REQUEST['id'], 'amf-' ) === 0
+			&& in_array( __NAMESPACE__ . '\\Traits\\RequestOne', class_uses( $provider ), true )
```

Comment:
This is not how traits should be used, and this will not work if the provider class is extending another class that uses the trait. What you're really testing for is whether the `request_item()` method exists.

We started going down the route of defining `supports_*()` methods in the base provider ([see here](https://github.com/humanmade/asset-manager-framework/blob/247ac39239a1c57e1d929d0888b4f0ee20e4129a/inc/Provider.php#L58-L88)) which would be overridden to return true in providers that support each feature. I think it would be better to use that here, for example by introducing a `supports_single_request()` method which returns false by default. This method can then be called here to check for support.

Reply:
```suggestion
			&& method_exists( $provider, 'request_item' )
```